check_all_obc_reached_bound_state_in_kube_job() fails with KeyError

In a bulk OBC creation when the OBC status field is not yet populated and empty check_all_obc_reached_bound_state_in_kube_job() fails with KeyError

```
kube_job_obj = <ocs_ci.ocs.resources.objectconfigfile.ObjectConfFile object at 0x7f66fb5a9a60>
namespace = 'stage-3-cycle-1', no_of_obc = 50, timeout = 60, no_wait_time = 20

    def check_all_obc_reached_bound_state_in_kube_job(
        kube_job_obj, namespace, no_of_obc, timeout=60, no_wait_time=20
    ):
        """
        Function to check either bulk created OBCs reached Bound state using kube_job
        Args:
            kube_job_obj (obj): Kube Job Object
            namespace (str): Namespace of OBC's created
            no_of_obc (int): Bulk OBC count
            timeout (second): a timeout for all the obc in kube job to reach bound state
            no_wait_time (int): number of wait time to ensure all OCBs to reach bound state
        Returns:
            obc_bound_list (list): List of all OBCs which is in Bound state.
        Raises:
            AssertionError: If not all OBC reached to Bound state
        """
        # Check all OBCs to reach Bound state
        obc_bound_list, obc_not_bound_list = ([] for i in range(2))
        while_iteration_count_1 = 0
        while_iteration_count_2 = 0
        while True:
            # Get kube_job obj and check if all OBC's are in Bound state
            # If not bound adding those OBCs to obc_not_bound_list
            job_get_output = kube_job_obj.get(namespace=namespace).get("items")
            if job_get_output is not None and len(job_get_output) == no_of_obc:
                for i in range(no_of_obc):
>                   status = job_get_output[i]["status"]["phase"]
E                   KeyError: 'status'

ocs_ci/ocs/scale_noobaa_lib.py:76: KeyError
```

--- a/ocs_ci/ocs/scale_noobaa_lib.py
+++ b/ocs_ci/ocs/scale_noobaa_lib.py
@@ -80,8 +80,13 @@ def check_all_obc_reached_bound_state_in_kube_job(
                     log.info(
                         f"obc {job_get_output[i]['metadata']['name']} status {status}"
                     )
-                except KeyError:
-                    if not status or status != constants.STATUS_BOUND:
+                except KeyError as err:
+                    msg = str(err)
+                    if (
+                        not status
+                        or "status" in msg
+                        or status != constants.STATUS_BOUND
+                    ):
                         obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])
                         # Wait 20 secs to ensure the next obc on the list has status field populated
                         time.sleep(30)

--- a/ocs_ci/ocs/scale_noobaa_lib.py
+++ b/ocs_ci/ocs/scale_noobaa_lib.py
@@ -73,13 +73,21 @@ def check_all_obc_reached_bound_state_in_kube_job(
         job_get_output = kube_job_obj.get(namespace=namespace).get("items")
         if job_get_output is not None and len(job_get_output) == no_of_obc:
             for i in range(no_of_obc):
-                status = job_get_output[i]["status"]["phase"]
-                log.info(f"obc {job_get_output[i]['metadata']['name']} status {status}")
-                if not status or status != constants.STATUS_BOUND:
-                    obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])
-                    # Wait 20 secs to ensure the next obc on the list has status field populated
-                    time.sleep(30)
-                    job_get_output = kube_job_obj.get(namespace=namespace).get("items")
+                try:
+                    # If the OBC status field is not yet populated and empty below line
+                    # throws KeyError. Handling it in try:except
+                    status = job_get_output[i]["status"]["phase"]
+                    log.info(
+                        f"obc {job_get_output[i]['metadata']['name']} status {status}"
+                    )
+                except KeyError:
+                    if not status or status != constants.STATUS_BOUND:
+                        obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])
+                        # Wait 20 secs to ensure the next obc on the list has status field populated
+                        time.sleep(30)
+                        job_get_output = kube_job_obj.get(namespace=namespace).get(
+                            "items"
+                        )
         else:
             while_iteration_count_1 += 1
             time.sleep(timeout)

--- a/ocs_ci/ocs/scale_noobaa_lib.py
+++ b/ocs_ci/ocs/scale_noobaa_lib.py
@@ -82,8 +82,8 @@ def check_all_obc_reached_bound_state_in_kube_job(
                     )
                 except KeyError as err:
                     if (
-                        not status
-                        or "status" in str(err)
+                        "status" in str(err)
+                        or not status
                         or status != constants.STATUS_BOUND
                     ):
                         obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])

--- a/ocs_ci/ocs/scale_noobaa_lib.py
+++ b/ocs_ci/ocs/scale_noobaa_lib.py
@@ -81,10 +81,9 @@ def check_all_obc_reached_bound_state_in_kube_job(
                         f"obc {job_get_output[i]['metadata']['name']} status {status}"
                     )
                 except KeyError as err:
-                    msg = str(err)
                     if (
                         not status
-                        or "status" in msg
+                        or "status" in str(err)
                         or status != constants.STATUS_BOUND
                     ):
                         obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])

